cmd/mayactl/app/command/volume_info.go

```diff
@@ -166,63 +172,64 @@ Status  :   {{.Status}}
 		fmt.Println("Error displaying volume details, found error :", err)
 		return nil
 	}
-	replicaCount, _ = strconv.Atoi(a.ReplicaCount)
-	// This case will occur only if user has manually specified zero replica.
-	if replicaCount == 0 || a.ReplicaStatus == "" || a.Replicas == "" {
-		fmt.Println("None of the replicas are running, please check the volume pod's status by running [kubectl describe pod -l=openebs/replica --all-namespaces] or try again later.")
-		return nil
-	}
-	// Splitting strings with delimiter ','
-	replicaStatusStrings := strings.Split(a.ReplicaStatus, ",")
-	addressIPStrings := strings.Split(a.Replicas, ",")
-
-	// making a map of replica ip and their respective status,index and mode
-	replicaIPStatus := make(map[string]*Value)
-
-	for i, v := range addressIPStrings {
-		if v != "nil" {
-			replicaIPStatus[v] = &Value{index: i, status: replicaStatusStrings[i], mode: "NA"}
-		} else {
-			// appending address with index to avoid same key conflict
-			replicaIPStatus[v+string(i)] = &Value{index: i, status: replicaStatusStrings[i], mode: "NA"}
+	if strings.Contains(v.Spec.Iqn, "com.openebs.jiva") {
+		replicaCount, _ = strconv.Atoi(v.Spec.Replicas)
+		// This case will occur only if user has manually specified zero replica.
+		if replicaCount == 0 || v.ObjectMeta.Annotations["openebs.io/replica-status"] == "" {
+			fmt.Println("None of the replicas are running, please check the volume pod's status by running [kubectl describe pod -l=openebs/replica --all-namespaces] or try again later.")
+			return nil
+		}
+		// Splitting strings with delimiter ','
+		replicaStatusStrings := strings.Split(v.ObjectMeta.Annotations["openebs.io/replica-status"], ",")
+		addressIPStrings := strings.Split(v.ObjectMeta.Annotations["openebs.io/replica-ips"], ",")
+
+		// making a map of replica ip and their respective status,index and mode
+		replicaIPStatus := make(map[string]*Value)
+
+		for i, v := range addressIPStrings {
+			if v != "nil" {
+				replicaIPStatus[v] = &Value{index: i, status: replicaStatusStrings[i], mode: "NA"}
+			} else {
+				// appending address with index to avoid same key conflict
+				replicaIPStatus[v+string(i)] = &Value{index: i, status: replicaStatusStrings[i], mode: "NA"}
+			}
 		}
-	}
 
-	// We get the info of the running replicas from the collection.data.
-	// We are appending modes if available in collection.data to replicaIPStatus
+		// We get the info of the running replicas from the collection.data.
+		// We are appending modes if available in collection.data to replicaIPStatus
 
-	replicaInfo := make(map[int]*ReplicaInfo)
-	for key := range collection.Data {
-		address = append(address, strings.TrimSuffix(strings.TrimPrefix(collection.Data[key].Address, "tcp://"), v1.ReplicaPort))
-		mode = append(mode, collection.Data[key].Mode)
-		if _, ok := replicaIPStatus[address[key]]; ok {
-			replicaIPStatus[address[key]].mode = mode[key]
+		replicaInfo := make(map[int]*ReplicaInfo)
+		for key := range collection.Data {
```

Comment:
I am not sure what are we trying to do here. 
Can we write modular functions or better this logic?